client/my-sites/subscribers/hooks/use-subscription-plans.ts

```diff
@@ -7,52 +7,66 @@ import {
 } from 'calypso/my-sites/earn/memberships/constants';
 import { Subscriber, SubscriptionPlan } from '../types';
 
-type SubscriptionPlanData = {
+export type SubscriptionPlanData = {
 	plan: ReactNode;
 	startDate?: string;
 	title?: string;
+	is_gift: boolean;
+};
+
+type PlanData = {
+	is_gift: boolean;
+	renewalPrice: string;
+	when: string;
+	start_date: string;
+	title: string;
 };
 
 const useSubscriptionPlans = ( subscriber: Subscriber ): SubscriptionPlanData[] => {
 	const translate = useTranslate();
 	const freePlan = translate( 'Free' );
 
-	const getPaymentInterval = ( renew_interval: string ) => {
+	const getPaymentInterval = ( renew_interval: string ): string => {
 		if ( renew_interval === null ) {
 			return translate( 'one time' );
 		} else if ( renew_interval === PLAN_MONTHLY_FREQUENCY ) {
 			return translate( 'Monthly' );
 		} else if ( renew_interval === PLAN_YEARLY_FREQUENCY ) {
 			return translate( 'Yearly' );
 		}
+
+		return '';
 	};
 
 	function formatRenewalPrice( renewalPrice: number, currency: string ) {
 		if ( ! renewalPrice ) {
 			return '';
 		}
-
-		const money = getCurrencyObject( renewalPrice, currency );
-		return money.integer !== '0' ? `${ money.symbol }${ money.integer }` : '';
+		const money = getCurrencyObject( renewalPrice, currency, { stripZeros: false } );
+		return money.hasNonZeroFraction
```

Comment:
That seems to fix @JuanLucha code and properly display plan prices with decimals (when needed), and prevent displaying an empty string if the price is < 0